src/lib/krb5/ccache/cc_mslsa.c

```diff
@@ -832,15 +833,18 @@ KerbSubmitTicket( HANDLE LogonHandle, ULONG  PackageId,
                                            NULL,
                                            &SubStatus
     );
+
+    rv = (!FAILED(Status) && !FAILED(SubStatus));
+
+cleanup:
     free(pSubmitRequest);
     if (keyblock)
         krb5_free_keyblock(context, keyblock);
+    if (krb_cred)
```

Comment:
Let's de-conditionalize the krb5_free_keyblock() and krb5_free_data() calls while we're introducing a cleanup label.  (Both functions can handle null pointers.)

Reply:
revision `ef0f174` fixes nits pointed out by Robbie and Greg